### ISSUE TITLE
Dropout op should be enabled

### ISSUE THREAD

**Issue by winston-zillow** [onnx/onnx-tensorflow#552]
The `Dropout` op is disabled due to these lines:
```python
    if cls.SINCE_VERSION >= 7 or attrs.pop("is_test", 0) == 1:
      return [x]
```
It appears a deliberated decision, perhaps because ONNX itself doesn't know what to do about the drop rate handling? This is a major blunder for trying to continuously train/fine-tune on existing imported model.

I suggest that `onnx-tensorflow` automatically generated a placeholder holding a boolean indicating whether it is in training and then it compute a dropout rate tensor based on the placeholder value. Here are the code skeleton inspired by [this post](https://stackoverflow.com/questions/44971349/how-to-turn-off-dropout-for-testing-in-tensorflow/44971517)

First add a command line option: `is_training_tensor_name`, default to `is_training`.

```python
class BackendHandler(Handler):
  @classmethod
  def get_or_create_tf_is_training(cls):
    is_training_tensor = command line option is_training_tensor_name
    with tf.name_scope(""):
      try:
        tf_is_training_float = tf.get_default_graph().get_tensor_by_name('%s_float:0' % is_training_tensor)
      except KeyError:
        tf_is_training = tf.placeholder_with_default(False, tuple(), name=is_training_tensor)
        tf_is_training_float = tf.cast(tf_is_training, tf.float32, name='%s_float' % is_training_tensor)
      return tf_is_training_float

class Dropout(BackendHandler):
  @classmethod
  def _common(cls, node, **kwargs):
    x = kwargs["tensor_dict"][node.inputs[0]]
    attrs = copy.deepcopy(node.attrs)
    # by default generate a placeholder with default value set to False, 
    # can skip the node by command-line option for small optimization
    if attrs.pop("is_test", 0) == 1 or option is_training_tensor_name is None
       return [x]
    rate = attrs.pop("ratio", 0.5)
    tf_is_training = cls.get_or_create_tf_is_training()
    return [tf.nn.dropout(x, rate=tf.multiply(tf_is_training, rate))]
```

**Comment by winston-zillow** [comment]
CC @winnietsang @chinhuang007

**Comment by winnietsang** [comment]
@winston-zillow currently there is no use case for the above usage, because ONNX-TF doesn't support training. We can revisit this once training capability is added to ONNX-TF.

**Comment by gautambathla** [comment]
@winnietsang, is it the case for both TF1.x and TF2.x? Also, why are all the variables converted to constants by ONNX during conversion? Is there a way to prevent this?

**Comment by shahirdaya** [comment]
I have created PR #671 for this.
